### ISSUE TITLE
Feature: server clause

### ISSUE THREAD

**Issue by NickNickNickNick** [WillyXJ/facileManager#218]
I am using fmDNS 2.0.1 and trying add a server clause (for TSIG) but not able to find it. I was able to add the key and allow-transfer option but not seeing the server clause.

key "tsig-key" {
    algorithm hmac-md5;
    secret " contents\of\secret\key";
};

allow-transfer {  key tsig-key; };

Should look something like this:

server 192.168.1.100 {
 keys { tsig-key; };
};
server 192.168.1.101 {
 keys { tsig-key; };
};
server 192.168.1.102 {
 keys { tsig-key; };
};

Could this be added to fmDNS?

Information on server clause: http://www.zytrax.com/books/dns/ch7/server.html
Information on TSIG: ftp://ftp.isc.org/isc/bind/cur/9.9/doc/arm/Bv9ARM.ch04.html#tsig

**Comment by WillyXJ** [comment]
Hi Nick,

The server clause isn't fully supported (yet), but you should be able to use the key feature. Once you define the key, go to Config->Servers, click the edit button for the server you want, and in the box you can select the key. Let me know if this doesn't achieve what you're looking for.

**Comment by NickNickNickNick** [comment]
Hey thank you for your reply. I have the keys feature working and it should get be by for now. Just wondering when server clauses will be fully supported in fmDNS?

**Comment by WillyXJ** [comment]
Good question and unfortunately there isn't a target date let alone target version yet. Still lots to do. I'll leave this issue open to track the feature.

**Comment by andrewhotlab** [comment]
It would be great to have this feature. We need to make fmDNS's managed servers transfer a zone from other's name server using TSIG, and we are still unable to do it. We are running fmDNS 3.1.2. Thanks!

**Comment by WillyXJ** [comment]
I've been stuck on this feature for quite some time because of the design and intention of the Config->Servers page.  That was intended to configure the fmDNS servers so the clause has very limited support.  To change that, it might be best to remove the key select from the server and create a new page for "remote" servers where the full servers clause would be supported.  Would that make sense or be the best approach to implement this feature?

**Comment by WillyXJ** [comment]
Another idea is to support fmDNS servers and remote servers all within the same page.  The configuration window would just be different based on a drop-down selection for "local" and "remote" servers.

The configuration window would, of course, include the supported server clause options.  This might be a better way to go for ease-of-use.

**Comment by andrewhotlab** [comment]
From a logic perspective, since the BIND `server` configuration statement can contains host-specific data (i.e. the `query-source` option), it would be better to include it in the Config -> Servers page, even if I guess that a dedicated page for "remote servers" would be easier to use.

I vote for the "easier-to-develop" way! :)

**Comment by WillyXJ** [comment]
This is now implemented in [fmDNS 3.3](http://www.facilemanager.com/modules/fmdns/) and later.

**Comment by andrewhotlab** [comment]
Wonderful! Thank you so much for this work: we upgraded to the latest release, defined a few "remote" servers and set for them TSIG keys... zone transfers from those server to the ones managed with fmDNS are now authenticated!

Just a question about the the count of remote servers that we noticed in the left menu, as if there where some change to commit (see attached screenshot): it's a normal behavior, or we misconfigured something?

![Screenshot_2019-03-18_11-36-16](https://user-images.githubusercontent.com/26433166/54525055-a3d19980-4973-11e9-98b0-6a277956088e.png)
